irnl_rdt_correction/io_handling.py

```diff
@@ -273,10 +274,20 @@ def _check_dfs(beams: Sequence[int], twiss_dfs: Sequence[DataFrame], errors_dfs:
                     raise IOError(text)
                 LOG.warning(text + " They are assumed to be zero.")
                 for kl in missing_cols:
-                    df[kl] = 0
+                    df[kl] = 0.0
     return beams, twiss_dfs, errors_dfs
 
 
+def convert_numeric_columns_to_float(df: TfsDataFrame) -> TfsDataFrame:
+    """ Convert numeric columns in df to float. 
+    This avoids that the user accidentally gives columns with dtype int (e.g. all 0), 
+    in which case assigning float values will fail in some pandas version after 2.1.0 
+    (which had a deprecation warning). """
+    numeric_columns = df.select_dtypes(include=['number']).columns
+    df[numeric_columns] = df[numeric_columns].astype('float64')
+    return df
```

Comment:
Would be good to at least specify in the docstring that this acts *inplace*. The function signature implies a new dataframe is returned, but this is not the case.

Reply:
look again! ( ;) )

Thanks! Well spotted!